addon/model-data.js

```diff
@@ -17,10 +18,38 @@ function commitDataAndNotify(modelData, updates) {
 class M3SchemaInterface {
   constructor(modelData) {
     this.modelData = modelData;
+    this._keyBeingResolved = null;
+    this._refKeyUnknownKeysMap = {};
+  }
+
+  _beginUnknownKeyResolution(key) {
+    assert('_keyBeingResolved should be null.', this._keyBeingResolved === null);
```

Comment:
Can we change the assertion message to be more relevant to the user rather than focusing on internal details?

Maybe something like

```
"Do not manually call methods on `schemaInterface` outside of schema resolution hooks such as `computeAttributeReference`"
```

what do you think?